src/metabase/sync/analyze/fingerprint/global.clj

```diff
@@ -0,0 +1,13 @@
+(ns metabase.sync.analyze.fingerprint.global
+  "Logic for generating a `GlobalFingerprint` from a sequence of values for a Field of *any* type."
+  (:require [metabase.sync.interface :as i]
+            [schema.core :as s]))
+
+(s/defn ^:always-validate global-fingerprint :- i/GlobalFingerprint
+  "Generate a fingerprint of global information for Fields of all types."
+  [values :- i/ValuesSample]
+  ;; TODO - this logic isn't as nice as the old logic that actually called the DB
```

Comment:
I think having the ability to run fingerprinting with a "full sample" is an eventual addition to this. 

I think we should try this out with actual datasets and see if it works. My hunch is that it'll be acceptable but we should verify.